pandas/core/internals/__init__.py

```diff
@@ -86,6 +86,55 @@
 from pandas.compat import range, map, zip, u
 
 
+def try_cast_result(left, result, dtype=None):
```

Comment:
IF you need this function, it should be in panda/core/dtypes/cast.py. I say IF because this looks like lots of other functions there. please see if you can simplify. we want to remove things from internals not add them.

Reply:
> it should be in panda/core/dtypes/cast.py

Sure, though I'm hoping we can get rid of it before long.

> I say IF because this looks like lots of other functions there

This is based on the existing `Block._try_cast_result`, which is changed to just call this function.

> we want to remove things from internals not add them.

Very much so.  The point of this PR (along with another branch in progress) is to remove `Block.eval` and `BlockManager.eval`. (and in doing so ensure that Series/DataFrame ops all agree)